cmd/sovereignnode/incomingHeader/incomingHeaderProcessor_test.go

```diff
@@ -412,128 +416,243 @@ func TestIncomingHeaderHandler_AddHeaderErrorCases(t *testing.T) {
 		require.ErrorIs(t, err, errMarshaller)
 		require.Equal(t, 0, numSCRsAdded)
 	})
+
+	t.Run("cannot create event data, should return error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createArgs()
+
+		args.DataCodec = &mock.DataCodecMock{
+			DeserializeEventDataCalled: func(_ []byte) (*sovereign.EventData, error) {
+				return nil, fmt.Errorf("cannot deserialize event data")
+			},
+		}
+
+		incomingHeader := &sovereign.IncomingHeader{
+			Header: &block.HeaderV2{},
+			IncomingEvents: []*transaction.Event{
+				{
+					Identifier: []byte(eventIDDepositIncomingTransfer),
+					Topics:     [][]byte{[]byte(topicIDDepositIncomingTransfer), []byte("addr"), []byte("tokenID1"), []byte("nonce1"), []byte("tokenData1")},
+					Data:       []byte("data"),
+				},
+			},
+		}
+
+		handler, _ := NewIncomingHeaderProcessor(args)
+		err := handler.AddHeader([]byte("hash"), incomingHeader)
+		require.ErrorContains(t, err, "cannot deserialize event data")
+	})
+
+	t.Run("cannot create token data, should return error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createArgs()
+
+		args.DataCodec = &mock.DataCodecMock{
+			DeserializeTokenDataCalled: func(_ []byte) (*sovereign.EsdtTokenData, error) {
+				return nil, fmt.Errorf("cannot deserialize token data")
```

Comment:
Same remark as above.
Also, I don't think you need `DeserializeEventDataCalled` below